docs(material/tree): add instructing to use flat tree for best perfor… (#25211)
angular/components: src/material/tree/tree.md

<!-- example(tree-flat-overview) -->

> Flat trees are generally easier to style and inspect. They are also more friendly to scrolling
> variations, such as infinite or virtual scrolling. Out of the two types of trees, flat tress
> has the best performance, and can render large data sets.

Reviewer: ```suggestion
Flat trees are generally easier to style and inspect. They are also more friendly to scrolling
variations, such as infinite or virtual scrolling. Flat trees
generally offer better performance.
```
